lib/unparser/emitter/literal/primitive.rb

```diff
@@ -119,7 +119,30 @@ def write_rational(value)
 
         end # Rational
 
-        # Emiter for numeric literals
+        # Emitter for strings
+        class String < self
+
+          handle :str
+
+        private
+
+          # Dispatch value
+          #
+          # @return [undefined]
+          #
+          # @api private
+          # 
+          def dispatch
+            if node.location.is_a?(Parser::Source::Map::Collection) && node.location.begin && node.location.end
```

Comment:
Unparser is supposed to reconstruct concrete syntax from the AST, not the source locations.

This is so you can fed it programatically constructed AST, that naturally does not have source locations.

If you wanted me to support the "emit single quotes" feature, you'd have to look at the containing string and verify it can be emitted with signle quotes. Not conditionally doing this in the presence of source locations.